# Describing Data

### 1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [15]:
import numpy as np

In [19]:
ages = [14,12,11,10,8,6,8]

#### Mean

In [16]:
#By Hand
mean = (14+12+11+10+8+6+8)/7
mean

9.857142857142858

In [17]:
#Numpy
np.mean(ages)

9.8571428571428577

#### Median

In [18]:
#By Hand
# Put ages in order: 6,8,8,10,11,12,14
10

10

In [21]:
#Numpy
np.median(ages)

10.0

#### Mode

In [23]:
#By Hand
8

8

In [25]:
#Statistics
import statistics
statistics.mode(ages)

8

#### Variance

In [36]:
#By Hand
ages = [14,12,11,10,8,6,8]
result=[]
for age in ages:
    result.append((age-mean)**2)
    
variance = sum(result)/6
variance

7.476190476190475

In [28]:
#Numpy
np.var(ages, ddof=1)

7.4761904761904754

#### Standard Deviation

In [37]:
#By Hand
std = variance**0.5
std

2.734262327610589

In [35]:
#Numpy
np.std(ages,ddof=1)

2.7342623276105891

#### Standard Error

In [38]:
#By Hand
SE = std/(7**0.5)
SE

1.0334540197243192

In [39]:
#Numpy
np.std(ages,ddof=1)/np.sqrt(len(ages))

1.0334540197243192

### 2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

Typically I would chose the mean because it accounts for all the data. I would do that in this case especially because there are no outliers that can skew the data. If there were outliers I would use the median. Furthermore, the mean and median and pretty close.

I would chose the Standard Error because its the margin of error.

### 3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [40]:
cindy_update_ages = [14,12,11,10,8,7,8]

In [43]:
#Mean
np.mean(cindy_update_ages)

10.0

In [44]:
#Median
np.median(cindy_update_ages)

10.0

In [45]:
#Variance
np.var(cindy_update_ages, ddof=1)

6.333333333333333

In [46]:
#STD
np.std(cindy_update_ages,ddof=1)

2.5166114784235831

In [47]:
#SE
np.std(cindy_update_ages,ddof=1)/np.sqrt(len(cindy_update_ages))

0.95118973121134176

The mean slightly increased, but the median and mode stayed the same.

Additionally, the variance, standard deviation and standard errors all decreased.

### 4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [49]:
oliver_sucks_ages = [14,12,11,10,8,7,1]

In [50]:
#Mean
np.mean(oliver_sucks_ages)

9.0

In [51]:
#Variance
np.var(oliver_sucks_ages, ddof=1)

18.0

In [52]:
#STD
np.std(oliver_sucks_ages,ddof=1)

4.2426406871192848

In [53]:
#SE
np.std(oliver_sucks_ages,ddof=1)/np.sqrt(len(oliver_sucks_ages))

1.6035674514745462

This does change my choice to use the median simply because the 1 year old is an outlier.

### 5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

### Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [56]:
magazines = [20,23,17,5]

In [57]:
np.mean(magazines)

16.25

In [59]:
np.std(magazines,ddof=1)/np.sqrt(len(magazines))

3.9449334595148748

I would estimate 16% +- 4% of adults were fans of the show. 